addons/mail/models/mail_activity.py

```diff
@@ -101,6 +103,19 @@ def _compute_delay_label(self):
             unit = selection_description_values[activity_type.delay_unit]
             activity_type.delay_label = '%s %s' % (activity_type.delay_count, unit)
 
+    @api.depends('default_next_type_id')
+    def _onchange_default_next_type_id(self):
+        """ This onchange is useful for the specific case of the activity type 'upload_file' :
+        Since the activity will be marked as done as soon as the file is uploaded from the selector
+        (without an opportunity to select a recommendation), the activity_type can either have the chaining_type
+        'recommend_next' when there is NO recommendation or have the chaining_type 'force_next' with a specified
+        default_next_type_id. The chaining_type cannot be selected in the form and is instead maintained here."""
+        for activity_type in self:
+            if activity_type.default_next_type_id and activity_type.chaining_type == 'recommend_next':
+                activity_type.chaining_type = 'force_next'
+            elif not activity_type.default_next_type_id and activity_type.chaining_type == 'force_next':
+                activity_type.chaining_type = 'recommend_next'
```

Comment:
The check on the current chaining type feels more confusing than necessary, just set the chaining type based on the next_type_id, it's way clearer, and a no-op update doesn't really matter even if the onchange system doesn't ignore it.

Reply:
Then wouldn't i have to display next_type_id in the form at all time ? (LNA) wanted this 'chaining_type' selection in order to allow alternating the display of "Force Next" and "Recommend Next". This onchange is only there to handle the specific case of `category='upload_file'` which will not allow the user to select a "recommended activity type" because it is integrated in an automated process. But I will think about another solution to avoid this onchange